### PR TITLE
fix: avatar loading issue

### LINKED ISSUE #8448: [CAL-1574] Avatar not Loading after New Account Setup
This is a dumby account, but the avatar isn't loading after it's been set in settings (granted username has not been paid for)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1574](https://linear.app/calcom/issue/CAL-1574/avatar-not-loading-after-new-account-setup)</sub>

### PR DESCRIPTION
**Pull request by Keyurkd**
## What does this PR do?

Fixes #8448 

## Type of change

- Bug fix:  Fixes an avatar not updating based on user profile update.

### CODE REVIEW

**Comment by Keyurkd** [comment]
@hariombalhara @Udit-takkar ,
Please review this.

File: apps/web/pages/event-types/index.tsx [hunk 1]
```diff
@@ -675,7 +679,11 @@ const EventTypeListHeading = ({
       <Avatar
         alt={profile?.name || ""}
         href={teamId ? `/settings/teams/${teamId}/profile` : "/settings/my-account/profile"}
-        imageSrc={`${WEBAPP_URL}/${profile.slug}/avatar.png` || undefined}
+        imageSrc={
+          loggedInUser?.username === profile.slug
```
Comment (hariombalhara) [inline apps/web/pages/event-types/index.tsx:683]: I don't understand the reasoning behind this change. `${WEBAPP_URL}/${profile.slug}/avatar.png` is supposed to serve the correct avatar. Why is it not working?

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Need clarification

File: apps/web/pages/event-types/index.tsx (hunk 1 above)
Comment (Keyurkd) [reply apps/web/pages/event-types/index.tsx:683]: @hariombalhara Yes `${WEBAPP_URL}/${profile.slug}/avatar.png` should have returned the correct avatar. It does return the correct avatar till the time user doesn't change the avatar in the current session. Due to session caching in the browser, it's not updating the image that the logged-in user has changed in the profile settings of a currently active session.

**Comment by Keyurkd** [comment]
@hariombalhara You can follow these https://github.com/calcom/cal.com/issues/8567 https://github.com/calcom/cal.com/issues/8610 issues which have been raised on the profile update for more information.

File: apps/web/pages/event-types/index.tsx (hunk 1 above)
Comment (Udit-takkar) [reply apps/web/pages/event-types/index.tsx:683]: @Keyurkd is there any way we can invalidate the avatar in the session cache when user updates it

Comment (hariombalhara) [reply apps/web/pages/event-types/index.tsx:683]: I don't think it is about the session. The problem is that the endpoint avatar.png is actually a [dynamic endpoint](https://github.com/calcom/cal.com/blob/5362cd66b4ab23392f1ad5ce5d5eca84aad0ab2a/apps/web/pages/api/user/avatar.ts#L68) that decides what's the profile pic for a user.  It currently has no cache-control headers and thus browsers are free to cache it. 

Making it completely non-cacheable will fix this issue but it would unnecessarily make function invocations for something that rarely changes.

This needs an RFC, but one quicker approach in our context can be that change the URL of the image every time the avatar changes. e.g. we compute and md5 of the image being uploaded and use that as query param in front of URL. Now, even if it is fully cached, changing the query param would be considered a different resources in terms of cache and will be fetched and cached again. We would have to store that md5 per user.

Also, if the issue is about the session not updating as well then there is a an update method available that might help https://next-auth.js.org/getting-started/client#updating-the-session

**Comment by zomars** [comment]
While this solution works fine. Can't we a find a way to invalidate `/[usename]/avatar.png` from the client instead? That way we could force refresh everywhere without minor code changes.

**Comment by Keyurkd** [comment]
@hariombalhara 

The `trigger` and `newSession` properties don't exist on NextAuth Options (`AUTH_OPTIONS`). Please refer to the below screenshots.

Screenshot 2023-05-12 at 4 00 56 PM


Screenshot 2023-05-12 at 4 03 27 PM


The `update` method doesn't exist in the useSession().data.

Screenshot 2023-05-12 at 4 07 07 PM

**Comment by PeerRich** [comment]
tldr: i think the avatar endpoint is the culprit. turning into draft
